scripts/schemas/board-schema.yml

```diff
@@ -78,3 +78,56 @@ mapping:
     type: seq
     sequence:
       - include: board-schema
+  run_once:
+    type: seq
+    desc: |
+          Allows for restricting west flash commands when using sysbuild to run once per given
+          grouping of boards. This is to allow for future image program cycles to not erase the
+          flash of a device which has just been programmed by another image.
+    sequence:
+      - type: map
+        mapping:
+          regex;(.*):
```

Comment:
this makes the command to become a key in the yaml file, not a value.

As I see it, there are two main implecations of such approach
- The free form `.*` means there is effectively no schema check, as anything is allowed for a value.
- A `:` must suddenly follow the `value`.
   In yaml, a value is either after the colon, like:
   `<key>: <value>`
   or a list of values, like:
   ```
   my_list:
   - value1
   - value2
   - value3
   - ...
   ```

but you're creating a list of maps, but where the `key` in that list is actually the value.

I don't see why regular `key: value` mapping cannot be used, and thereby avoid tricks like this.

Reply:
Unification with snippets

Reply:
but not unified with boards.yml.

For example, the board.yml for nrf9160dk does:
```
board:
  name: nrf9160dk
  vendor: nordic
  socs:
  - name: nrf9160
    variants:
    - name: 'ns'
  - name: nrf52840
```

and **not**:
```
board:
  name: nrf9160dk
  vendor: nordic
  socs:
  - nrf9160:
      variants:
      - name: 'ns'
  - nrf52840:
```

The fact that snippet does this does not justify this principle in board or soc yaml files.

Especially not when this proposal introduces inconsistency on how things are done within the same yaml file.

Reply:
See also comment here: https://github.com/zephyrproject-rtos/zephyr/pull/69748#discussion_r1530031405